app/controllers/posts_controller.rb

```diff
@@ -387,13 +387,13 @@ def delete
       return
     end
 
-    if @post.post_type.is_freely_editable && !current_user&.is_moderator
+    if @post.post_type.is_freely_editable && !current_user&.at_least_moderator?
       flash[:danger] = helpers.i18ns('posts.cant_delete_community')
       redirect_to post_path(@post)
       return
     end
 
-    if @post.children.any? { |a| !a.deleted? && a.score >= 0.5 } && !current_user&.is_moderator
+    if @post.children.any? { |a| !a.deleted? && a.score >= 0.5 } && !current_user&.at_least_moderator?
```

Comment:
Untested suggestion:

```suggestion
    if @post.children.any? { |a| !a.deleted? && a.score >= 0.5 } && current_user&.standard?
```

Reply:
As per our discussion, let's leave this one be as well, as `!current_user&.at_least_moderator?` is actually equivalent to `!current_user || current_user.standard?` (safe navigation operator)